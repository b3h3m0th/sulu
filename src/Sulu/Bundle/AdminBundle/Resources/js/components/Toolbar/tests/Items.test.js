/* eslint-disable flowtype/require-valid-file-annotation */
import {render} from '@testing-library/react';
import React from 'react';
import debounce from 'debounce';
import Items from '../Items';
import Button from '../Button';

const clickSpy = jest.fn();

jest.mock('debounce', () => jest.fn((callback) => callback));

window.ResizeObserver = jest.fn(function() {
    this.observe = jest.fn();
    this.disconnect = jest.fn();
});

test('Render items', () => {
    const {container} = render(<Items />);
    expect(container).toMatchSnapshot();
});

test('Render items with children', () => {
    const {container} = render(<Items><Button onClick={clickSpy}>Test</Button></Items>);
    expect(container).toMatchSnapshot();
});

test('Resize div should call callback', () => {
    const resizeFunction = jest.fn();
    debounce.mockReturnValue(resizeFunction);

    const {container} = render(
        <Items>
            <Button>Test</Button>
        </Items>
    );

    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const items = container.querySelector('.itemsContainer');

    expect(ResizeObserver).toBeCalledWith(resizeFunction);
    expect(ResizeObserver.mock.instances[0].observe).toBeCalledWith(items);
    expect(items).toHaveTextContent('Test');
});

test('ResizeObserver.disconnect should be called before component unmount', () => {
    const {unmount} = render(
        <Items>
            <Button>Test</Button>
        </Items>
    );
    unmount();

    expect(ResizeObserver.mock.instances[0].disconnect).toBeCalled();
});
