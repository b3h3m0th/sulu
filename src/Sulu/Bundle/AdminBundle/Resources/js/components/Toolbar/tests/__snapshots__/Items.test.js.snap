// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render items 1`] = `
<div>
  <div
    class="itemsContainer"
  >
    <ul
      class="items light"
    />
  </div>
</div>
`;

exports[`Render items with children 1`] = `
<div>
  <div
    class="itemsContainer"
  >
    <ul
      class="items light"
    >
      <li>
        <button
          class="button light"
          type="button"
        >
          <span
            class="label"
          >
            Test
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
